plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/config.py

```diff
@@ -14,13 +14,13 @@ class BaseQueueConf:
     # maximum time for the job in minutes
     timeout_min: int = 60
     # number of cpus to use for each task
-    cpus_per_task: int = 1
+    cpus_per_task: Optional[int] = None
     # number of gpus to use on each node
-    gpus_per_node: int = 0
+    gpus_per_node: Optional[int] = None
     # number of tasks to spawn on each node
     tasks_per_node: int = 1
     # memory to reserve for the job on each node (in GB)
-    mem_gb: int = 4
+    mem_gb: Optional[int] = None
```

Comment:
If those parameters should not always be passed, why are they in the Base class instead of in the subclass SlurmQueueConf?

Reply:
That's a good question, I'm happy to move them if it makes sense but we can wait to hear from @jrapin tomorrow to see if a reorg makes sense

Reply:
Let's do it and see if everything holds.
Once we do - we can eliminate the None filtering. (None can be a perfectly legitimate value of some future parameter).

Reply:
I think we still need the none filtering because it's AutoExecutor that has the issue and everything ends up there. So even if we move to slurm it still goes through auto which complains if mem_gb is set to None instead of an int (but works fine if we just don't pass mem_gb). 

Basically the issue is that submitit is designed to have optional values be not passed at all vs assigned None

Reply:
So maybe AutoExecutor should be fixed to accept none for mutually exclusive parameters and do the proper validation internally?

Reply:
The issue with that is that the validation is done in the slurmexecutor already and would need to be duplicated in auto. We could make auto less picky about params and delegate that authority to the specific executor that gets used however. I think @jrapin didn't like the idea and indicated that the missing-value-instead-of-None thing was an intentional decision, so let's wait for his feedback

Reply:
So I am a bit confused.
If AutoExecutor is in the business of validating the input, it should be okay for it to apply specific logic to mutually exclusive fields.
If it's not, it should also not care if any of the input fields are None and should just pass them as is to the underlying executor.

Happy to wait for some feedback from jrapin.

Reply:
My understanding is that the AutoExecutor does two things

1. Route the call to the correct executor
2. Remove executor prefixes (like "slurm_") so that parameter names are correct.

It also seems to make sure that certain parameters which are considered shared are the correct type that the underlying executor expects but does no further validation. 

It could be argued that the type checking is simply incorrect since, using mem as an example, it checks for int but the slurm executor expects an optional int

Reply:
`BaseQueueConf`  contains the parameters that are shared between different executors, and supported by all of the executors, specific configurations derive from it because they are build on top of these shared parameters, adding different options.

> It could be argued that the type checking is simply incorrect since, using mem as an example, it checks for int but the slurm executor expects an optional int

mem is str, but it checks `mem_gb` is int, which it should be, so nothing incorrect here.

> If AutoExecutor is in the business of validating the input, it should be okay for it to apply specific logic to mutually exclusive fields.

It does not validate the inputs, slurm does, we don't want to duplicate validation.

> Basically the issue is that submitit is designed to have optional values be not passed at all vs assigned None

Exactly. So far the behavior of submitit is that if those base parameters are passed, they shoul be used. This behavior *could* be changed though so that `None` are ignored, I can check what Guillaume thinks of it. In any case, if you want to avoid incompatibilities I would advise to filter as well, at least for the time being.

Reply:
OK and as for 

> None can be a perfectly legitimate value of some future parameter

Is it safe to assume, for now, that `None` will not be a valid value or will be equivalent in meaning to not passing the parameter?

Reply:
In all other places in Hydra None is a legitimate value that can be used in configs (null in yaml).
The philosophy of Hydra that the config should be exactly what is needed by the app (to the possible extent).
Manipulating the config in code is not as flexible as composing the right thing in the first place.

I don't mind making None mean something special in this plugin but I expect it will confuse users of the plugin.
If we are planning on making a change in AutoExecutor anyway I would just wait for it. You are of course free to use your local hacked version of the plugin in the mean time.

Reply:
I'm not clear on how we would make it work otherwise

The way I'm looking at it is just that `None` in hydra `==` missing value in submitit, so it's just an API translation.

Reply:
> I don't mind making None mean something special in this plugin but I expect it will confuse users of the plugin.
If we are planning on making a change in AutoExecutor anyway I would just wait for it. You are of course free to use your local hacked version of the plugin in the mean time.

If we fix it in AutoExecutor, that's fine and I can rebase this PR to one that just adds in the new parameters and makes the old ones optional.

Reply:
With facebookincubator/submitit#1607 merged I'll update this PR to remove the None filtering

Reply:
I think I can't make a push on that change until submitit makes a release with facebookincubator/submitit#1607 is that correct @omry ?

Reply:
Hard to say, but it's easier to just do it once they release an updated version.